conda_declarative/edit.py

```diff
@@ -1,62 +1,324 @@
+        self.subdirs = subdirs
+
+        with open(self.filename) as f:
+            text = f.read()
+
+        self.saved_text = text
+        self.editor = TextArea.code_editor(
+            text=text,
+            language="toml",
+        )
+        self.output = DataTable()
+        self.output_data = []
+
+        super().__init__()
+
+    async def on_text_area_changed(self, event: TextArea.Changed) -> None:
+        """Run post change actions on the text.
+
+        First, we check for changes compared to the version on disk.
+        If there are changes:
+
+        - Update the title of the app
+        - Check that the text area contains valid toml.
+            - If valid:
+                - Run the solver to get the new dependencies
+                - Update the right hand pane with the new dependencies
+            - Otherwise, make a notification that the toml is invalid
+
+        Parameters
+        ----------
+        event : TextArea.Changed
+            Event that triggered the call
+        """
+        if event.text_area.text == self.saved_text:
+            self.title = f"Editing {self.filename}"
+        else:
+            self.title = f"Editing {self.filename} (Unsaved)"
+
+        self.output.loading = True
+        self.run_worker(self.update_table(), exclusive=True)
+
+    async def update_table(self, debounce: int = 1) -> None:
+        """Update the table with the solution associated with the current manifest."""
+        # Since this function runs inside an exclusive worker, this await serves to
+        # debounce input to avoid solving on every keypress.
+        if debounce > 0:
+            await asyncio.sleep(debounce)
+        try:
+            manifest = await asyncio.to_thread(loads, self.editor.text)
+        except Exception as e:
+            self.notify(
+                f"The current file is invalid TOML: {e}", severity="error"
+            )
+            return
+
+        # Store the current data so that we know which rows to update
+        current_data = {}
+        for row in self.output.rows:
+            name, version, build, build_number, channel = self.output.get_row(row)
+            current_data[name] = [version, build, build_number, channel]
+
+
+        with set_conda_console():
+            records = await asyncio.to_thread(
+                solve,
+                prefix=self.prefix,
+                channels=manifest.get("channels", []),
+                subdirs=self.subdirs,
+                specs=manifest.get("requirements", []),
+            )
+
+        rows = []
+        for record in records:
+            rows.append(
+                (
+                    record.name,
+                    record.version,
+                    record.build,
+                    record.build_number,
+                    record.channel,
+                )
+            )
+
+        self.output.clear()
+        self.output.add_rows(rows)
+        self.output.sort('name')
```

Comment:
Can the user sort the TUI table by clicking on columns? 🤔 They feel clickable in the UI but not sure if this is supposed to work or just something we didn't implement yet.

Reply:
Good idea. I added this ability, with indicators to show the sort order.